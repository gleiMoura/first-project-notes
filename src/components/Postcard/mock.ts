import { faker } from "@faker-js/faker"
import { Post } from "../../interface";

const mock: Post = {
    id: faker.number.int(),
    userId: faker.number.int(),
    title: faker.lorem.sentence(),
    body: faker.lorem.paragraph(),
    cover: faker.image.url()
}

export default mock;