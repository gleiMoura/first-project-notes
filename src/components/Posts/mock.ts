import { Post } from "../../interface";
import createMock from "../Postcard/mock";

const post_1: Post = createMock();
const post_2: Post = createMock();
const post_3: Post = createMock();

const posts: Post[] = [post_1, post_2, post_3];

export default posts;