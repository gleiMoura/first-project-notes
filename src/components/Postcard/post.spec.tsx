import { render, screen } from "@testing-library/react"
import { PostCard } from ".";
import mock from "./mock";

describe("<PostCard />", () => {
    it('Should render post card correctly', () => {
        render(<PostCard post={mock} />);

        expect(screen.getByRole('img', { name: mock.title }))
            .toHaveAttribute('src', mock.cover);
        expect(screen.getByText(mock.body))
            .toBeInTheDocument();
        expect(screen.getByText(mock.title))
            .toBeInTheDocument();
    });

    it('Should match snapshot', () => {
        const { container } = render(<PostCard post={mock} />);
        expect(container.firstChild).toMatchSnapshot();
    })
})