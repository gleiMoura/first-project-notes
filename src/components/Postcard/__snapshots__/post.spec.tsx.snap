// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> Should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Voluptate labore quod suscipit pariatur sequi soluta doloribus."
    src="https://loremflickr.com/640/480?lock=6662456810143744"
  />
  <div
    class="post-content"
  >
    <h3>
      Voluptate labore quod suscipit pariatur sequi soluta doloribus.
    </h3>
    <h5>
      Nemo porro suscipit velit asperiores nemo velit voluptates officia repellat. Exercitationem quidem alias. Tempora quidem distinctio itaque.
    </h5>
  </div>
</div>
`;
